syft/pkg/cataloger/apkdb/parse_apk_db.go

```diff
@@ -162,3 +182,42 @@ func parseApkDBEntry(reader io.Reader) (*pkg.ApkMetadata, error) {
 
 	return &entry, nil
 }
+
+func discoverPackageDependencies(pkgs []*pkg.Package) (relationships []artifact.Relationship) {
+	// map["provides" string] -> packages that provide the "p" key
+	lookup := make(map[string][]*pkg.Package)
+	// read "Provides" (p) and add as keys for lookup keys
+	for _, p := range pkgs {
+		apkg, ok := p.Metadata.(pkg.ApkMetadata)
+		if !ok {
+			// TODO warn
```

Comment:
we don't want to stop processing here, but we should log that something unexpected happened here